lib/eslint/eslint.js

```diff
@@ -1237,7 +1237,10 @@ class ESLint {
 			formatter = (await import(pathToFileURL(formatterPath))).default;
 		} catch (ex) {
 			// check for formatters that have been removed
-			if (removedFormatters.has(name)) {
+			if (
+				removedFormatters.has(name) &&
+				ex.code === "ERR_MODULE_NOT_FOUND"
+			) {
```

Comment:
As for the solution, at this point we already know whether we are trying to load a built-in formatter or an external module.

https://github.com/eslint/eslint/blob/2584187e4a305ea7a98e1a5bd4dca2a60ad132f8/lib/eslint/eslint.js#L1211-L1232

Somewhere in the above code we could set a flag that we're trying to load a built-in module, and only in that case throw an error with the `... formatter is no longer part of core ESLint` message. I think that would be more reliable than error code.